filesystems/xfs/xfstests/README.md

```diff
@@ -0,0 +1,22 @@
+# xfstests test suite
+Wrapper for xfstests test suite, see http://git.kernel.org/?p=fs/xfs/xfstests-dev.git;a=summary \
+Test Maintainer: [Xiong Murphy Zhou](mailto:xzhou@redhat.com)
+
+## How to run it
+
+### Configure which test cases to run, e.g.
+```bash
+$ export TEST_PARAM_FSTYPE=xfs
+$ export TEST_PARAM_RUNTESTS=generic/001
+```
+
+### Dependencies
+Dependencies will be automatically installed with dep-install.sh when you execute \
+'make run', however to see a complete list of dependencies, please see \
+https://git.kernel.org/pub/scm/fs/xfs/xfstests-dev.git/tree/README
+
+
+### Execute the test
+```bash
+$ make run
```

Comment:
I tried this on a fresh Fedora 29 workstation install and I got this:
```
[nkondras@localhost xfstests]$ make run
Makefile:32: /usr/share/rhts/lib/rhts-make.include: No such file or directory
make: *** No rule to make target '/usr/share/rhts/lib/rhts-make.include'.  Stop.
```

This obviously needs Beaker stack prerequisites installed. If we're targeting internal users so far, then just mentioning that could be good enough. However, if we want anybody else from the outside to run the test like you specify here, then we need to make sure they can get everything required, and can *actually* run the tests.

If we want to target outside developers, then probably the most economical way (and closest to what we run) would be to direct them to run the Beaker wrapper for every test, and describe how to install the dependencies. An easy way to do this is to get a fresh Fedora install, download the tests, try to run, and document everything you had to do to make them work.

I actually don't see any other way to do that, since we're downloading upstream code and are patching it. So to get what we run they should do the same.

Reply:
I completely forgot about restraint-rhts, I provisioned a system internally in beaker to run xfstests, and stripped it down to remove any restraint/beaker rpms and corresponding repos, then I did the following:
```
wget -O /etc/yum.repos.d/beaker-client.repo https://beaker-project.org/yum/beaker-client-Fedora.repo
wget -O /etc/yum.repos.d/beaker-harness.repo https://beaker-project.org/yum/beaker-harness-Fedora.repo
dnf install -y beaker-client beakerlib restraint-rhts
```
However it complained about conflicts and other failures, I must be missing other dependencies/repos. @jencce what other steps are we missing here?

Reply:
@rasibley, just install a Fedora 29 into a VM (e.g. in virt-manager) on your own host (it doesn't take long) and try to run tests there. This way you'll get cleanest instructions.

Reply:
Thanks @spbnick I did originally test on a vm using 1MinuteTip, but didn't think that was the right approach since I don't believe xfstests will run properly in a vm, maybe @jencce can confirm ?

Reply:
true :-) here is the complete log in a fresh VM install of F29 using virt-manager: https://paste.fedoraproject.org/paste/FpLt30acTogm6SubJVGrxw

I'll let @jencee confirm if we're missing anything.

Reply:
xfstests can run in VM well and Racheal's log looks good enough.

Reply:
thanks @jencce I updated the root README with the common steps, @spbnick can you take a look ? :-)